tests/ui.rs

```diff
@@ -118,7 +126,10 @@ fn ui_test_usage_help_flag() {
 #[allow(unused)]
 #[macro_export]
 macro_rules! ignore {
-    ($expr:expr) => {{
-        $expr
+    ($value:expr) => {{
+        $value
+    }};
+    ($name:expr, $value:expr) => {{
+        $value
     }};
 }
```

Comment:
How is this terrible code if it does what it is meant to do and it's the simplest way of doing so? :rofl:

---

Now, seriously, from the context of your comment, I assume you disagree with the decision (that I made) to disable these specific tests on Windows.

https://github.com/ouch-org/ouch/pull/555#issuecomment-1826862726

It takes work, and I simply don't have a Windows machine or the time and patience to take the CI feedback loop to support Windows, a platform where most people don't even use the CLI (not without WSL).

Ouch haven't been properly maintained because I don't have enough time for it, and Windows is close to the bottom of the list of priorities.

For Windows support I'm just accepting PRs, I'm not actively maintaining, that's a  commitment boundary.

---

Please, choose appropriate adjectives :+1:.

Reply:
- Does `insta` decently support snapshot tests with conditional compilation?
- How can I update the snapshot test results with Insta without a Windows machine?

Reply:
> How is this terrible code if it does what it is meant to do and it's the simplest way of doing so? 🤣

Pointing an alias to different macros conditionally is a *really* non-standard way to disable tests, not to mention it denies the opportunity to enable/disable the tests in file one by one. By "terrible code" all I meant was "there's better ways to do it"; please don't take offence to that.

> Now, seriously, from the context of your comment, I assume you disagree with the decision (that I made) to disable these specific tests on Windows.

No, that was not a rhetorical question. I just find it strange that a project would have a CI setup for Windows but also have tests disabled for this target, and I wanted to know why (and maybe try to fix the underlying cause). But thanks for explaining your rationale anyways, I understand why now.

Reply:
> * Does `insta` decently support snapshot tests with conditional compilation?

So I've dabbled with `insta` a fair bit in [bandwhich](https://github.com/imsnif/bandwhich), and I think I've got a fair understanding of how it works. For starters `insta` really isn't "conditional-compilation-aware" or even "target-aware". All it does is format/serialise the expression you pass in, generate a file name automatically using the test's module path if necessary, and save it as a file (what it calls a "snapshot"). Then you commit the snapshot, and for later test runs it compares the outputs to the existing snapshot. There is an example in this PR where I conditionally (switched on a feature flag) to specify a different name for the snapshot.

> * How can I update the snapshot test results with Insta without a Windows machine?

Unless the expected output of the expressions are different on Windows (which I don't believe is the case), you wouldn't need to, since it's reading from and comparing to the same snapshot files for all targets.